Data/ByteString/Streaming.hs

```diff
@@ -1650,7 +1650,7 @@ interact f = stdout (f stdin)
 -- {-# NOINLINE moduleError #-}
 
 revNonEmptyChunks :: [P.ByteString] -> ByteString m r -> ByteString m r
-revNonEmptyChunks = Prelude.foldr (\bs f -> Chunk bs . f) id
+revNonEmptyChunks = L.foldl' (\f bs -> Chunk bs . f) id
```

Comment:
The main significant change. `foldl` is what reverses lists, not `foldr`. 

**Question:** Is the strict variant what we want here? I haven't seen a time where the (default) lazy one is desired, but since the result is itself a function (`ByteString m r -> ByteString m r`) it's hard to say.

Reply:
The fold is only strict in data we already have in hand, the chunk fragments known to contain runs of characters from the group. So strictness is OK, but not actually useful, because composition of functions does not simplify until you supply an argument.

Thus, given e.g. [s3, s2, s1]  (the collected out of order chunks to use], with `foldl'` we get:
`(Chunk s1 . (Chunk s2 . (Chunk s3 . id )))`, which is surely the same thing we get with `foldl`.
When this is applied to a tail `t` we end up with: `Chunk s1 ((Chunk s2 . (Chunk s3 . id )) t)` which given that `Chunk is lazy in its second argument` may allow us to consume `s1` before dealing with the inner closure...  I would expect that the same happens with either `fold` or `foldl'`.  But I am willing to learn if that's a mistaken impression...